src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRunCodegenNodeFactory.cs

```diff
@@ -110,6 +110,11 @@ private ISymbolNode CreateReadyToRunHelperCell(ReadyToRunHelper helperId)
             SignatureContext signatureContext, 
             bool isUnboxingStub = false)
         {
+            if (targetMethod.HasCustomAttribute("System.Runtime.CompilerServices", "IntrinsicAttribute"))
```

Comment:
BTW: MethodDesc has `IsIntrinsic` property that is more efficient way to check whether the method should be treated as intrinsic by the codegen.

Reply:
Other than that, I was under the impression that JIT should expand an intrinsic by itself once it sees it. In this sense calling MethodEntrypoint for an intrinsic is kind of a pathological situation - perhaps I'm just missing the background to understand how such cases are supposed to get resolved.

Reply:
> according to my understanding it performs a text check on the simple assembly name to match either "System.Numerics" or "System.Numerics.Vectors"

The `Vector<T>` lives in CoreLib in .NET Core. The path that matters for .NET Core is where the `Vector<T>` is looked up in CoreLib. Feel free to submit PR to CoreCLR to delete the path that looks for `Vector<T>` in "System.Numerics" or "System.Numerics.Vectors" where it used to be in the past.

Note that crossgen refuses to compiler anything that has to do with `Vector<T>` type, not just method calls on it. For example, somebody can have `sizeof(System.Numerics.Vector<int>)` in the code, with no method calls on Vector. R2R compiler has to skip this code because of the runtime size of `Vector<int>` is not known.

> I was under the impression that JIT should expand an intrinsic by itself once it sees it.

The JIT is not required to expand an intrinsic. Expanding intrinsic is an optimization and the JIT is free to decide whether the optimization is benefical. The EE can ask the JIT to require to expand the intrinsic by setting `*pMustExpand` to true, however this flag is not supported for all intrinsics.